src/core/eventifier.js

```diff
@@ -182,6 +197,30 @@ function eventifier(target) {
         return eventHandlers[ns][name][type];
     }
 
+    /**
+     * Add a handler for one or more events of a given event type
+     * @param {String} eventNames - the name of the event, or multiple events separated by a space
+     * @param {String} type - the type of event in before, between and after
+     * @param {Function} handler - the callback to run once the event is triggered
+     * @param {Number} [handler.priority] - value which can be used to schedule the handler earlier or later in its sequence
+     */
+    function addHandler(eventNames, type, handler) {
+        if (_.isFunction(handler)) {
+            _.forEach(getEventNames(eventNames), eventName => {
+                if (typeof handler.priority !== 'number') {
+                    handler.priority = defaultPriority;
+                }
+                // insert to keep the handlers array sorted by decreasing priority (high -> low)
+                const handlers = getHandlers(eventName, type);
+                let insertionIndex = handlers.findIndex(h => h.priority < handler.priority);
+                if (insertionIndex === -1) {
+                    insertionIndex = handlers.length;
+                }
+                handlers.splice(insertionIndex, 0, handler);
```

Comment:
This is not enough, because `triggerBefore/triggerHandlers` still executes all handlers in parallel (`Promise.all(pHandlers)`).

But it should do `Promise.all` only for handlers of the same priority, and also not execute next-priority if higher-priority cancelled this event.

As for who sets priority, I think we can start from hardcoding it in customer plugin, but logically test-runner should manage that, maybe by updating config option (+ inside some `dynamicModulesIndex` util to allow customer repo override).
```
---- test-runner qti.js install/init
<checks which plugins are enabled>
config.options.plugin1.beforeMoveHandlerPriority = 0;
config.options.plugin2.beforeMoveHandlerPriority = 1;

---- plugin-1
handler.priority = config.options.plugin1.beforeMoveHandlerPriority || 0
.before('move', handler)

---- plugin-2
handler.priority = config.options.plugin2.beforeMoveHandlerPriority || 0
.before('move', handler)
```